lib/audienceMetrics.ts

```diff
@@ -24,11 +25,13 @@ export const getAudienceMetrics = async ({ socket }: { socket: WebSocket }) => {
         },
       );
 
+    await audienceMetricsCall.headers;
+    serverErrorSignal.value = "";
+
     for await (const response of audienceMetricsCall?.responses) {
       for (const [key, value] of Object.entries(response?.rates)) {
-        !response?.Keepalive && socket.send(
-          JSON.stringify({ [key]: value as RateMetrics }),
-        );
+        !response?.Keepalive &&
```

Comment:
Broke out socket send here to distinguish socket errors from server connection errors.